oval_graph/client.py

```diff
@@ -182,83 +182,116 @@ def get_src(self, src):
+                self.START_OF_FILE_NAME + rule + '.html')
         return os.path.join(
             os.getcwd(),
-            'graph-of-' + rule + '.html')
+            self.START_OF_FILE_NAME + rule + '.html')
 
     def _get_part(self, part):
+        out = ''
         with open(os.path.join(self.parts, part), "r") as data_file:
-            return data_file.readlines()
-
-    def _merge_report_parts(self, data):
-        head = self._get_part('head.txt')
-        css = self._get_part('css.txt')
-        boot_strap_style = self._get_part('bootstrapStyle.txt')
-        jsTree_style = self._get_part('jsTreeStyle.txt')
-        jQuery_script = self._get_part('jQueryScript.txt')
-        boot_strap_script = self._get_part('bootstrapScript.txt')
-        jsTree_script = self._get_part('jsTreeScript.txt')
-        body_start = ['</head>', '<body>', data]
-        body = self._get_part('body.txt')
-        script = self._get_part('script.js')
-        footer = ['<script>', *script, '</script>', '</body>', '</html>']
-        return [
-            *head,
-            *css,
-            *boot_strap_style,
-            *jsTree_style,
-            *jQuery_script,
-            *boot_strap_script,
-            *jsTree_script,
-            *body_start,
-            *body,
-            *footer]
-
-    def save_html_report(self, dict_, src, status_all_in_one=False):
-        data = ("\n<script>var all_in_one = " +
-                str(status_all_in_one).lower() +
-                ";</script>" +
-                "\n<script>var data_of_tree =" +
-                str(json.dumps(dict_, sort_keys=False, indent=4)) +
-                ";</script>\n")
-        if status_all_in_one:
-            for rule in dict_:
-                data += ('<h1>' +
-                         rule.split('-')[2] +
-                         '</h1>' +
-                         '<div id="' +
-                         re.sub(r'[\_\-\.]', "", rule) +
-                         '"></div>')
+            for line in data_file.readlines():
+                out += line
+            return out
+
+    def _get_html_head(self):
+        return E.head(
+            E.title("OVAL TREE"),
+            E.style(self._get_part('css.txt')),
+            E.style(self._get_part('bootstrapStyle.txt')),
+            E.style(self._get_part('jsTreeStyle.txt')),
+            E.script(self._get_part('jQueryScript.txt')),
+            E.script(self._get_part('bootstrapScript.txt')),
+            E.script(self._get_part('jsTreeScript.txt')),
+        )
+
+    def _get_html_body(self, dict_of_rules):
+        return E.body(
+            E.script(self._get_script_graph_data(dict_of_rules)),
+            self._get_titles_and_places_for_graph(dict_of_rules),
+            E.div({'id': 'data'}),
+            E.div({'id': 'modal', 'class': 'modal'},
+                  E.div({'class': 'modal-content'},
+                        E.span({'id': 'closer', 'class': 'close'}, '&times;'),
```

Comment:
I don't understand why, but I guess that this is not correct - when I click rule details, there is not the "X", but literal `&times;` that doesn't close the window if clicked.